system/src/system_control_internal.cpp

```diff
@@ -258,7 +364,279 @@ void particle::SystemControl::processRequest(ctrl_request* req, ControlRequestCh
         setResult(req, r);
         break;
     }
+    case CTRL_REQUEST_WIFI_SCAN:
```

Comment:
I would prefer these handlers to be broken out into separate functions. It makes reading a lot easier and the compilers job easier too, leading to potentially smaller code.

Reply:
These are already refactored in #1444. See https://github.com/particle-iot/firmware/tree/a3b09a11bf71cbf9cacb8961578ca9de55bb14d6/system/src/control

Reply:
sweet!